distributed/tests/test_worker_memory.py

```diff
@@ -971,19 +971,21 @@ def __reduce__(self):
             return bool, (paused,)
 
     futs = c.map(SlowSpill, range(N_TOTAL))
-    while len(a.data.slow) < (N_PAUSE + 1 if a.state.ready else N_PAUSE):
+    while not a.data.slow:
         await asyncio.sleep(0.01)
 
     assert a.status == Status.paused
     # Worker should have become paused after the first `SlowSpill` was evicted, because
     # the spill to disk took longer than the memory monitor interval.
     assert len(a.data.fast) == 0
+
+    await c.gather(futs)
     # With queuing enabled, after the 3rd `SlowSpill` has been created, there's a race
     # between the scheduler sending the worker a new task, and the memory monitor
     # running and pausing the worker. If the worker gets paused before the 4th task
     # lands, only 3 will be in memory. If after, the 4th will block on the semaphore
     # until one of the others is spilled.
-    assert len(a.data.slow) in (N_PAUSE, N_PAUSE + 1)
+    assert len(a.data.fast) == 0
```

Comment:
It's interesting. This currently fails on CI because apparently *no* data was spilled

```
>       assert len(a.data.fast) == 0
E       AssertionError: assert 5 == 0
```